app/assets/javascripts/Grader/marking.js

```diff
@@ -99,23 +101,23 @@ jQuery(document).ready(function() {
   // Handle the expand/collapse buttons
   jQuery('#expand_all').click(function() {
     jQuery('.mark_criterion_level_container').each(function() {
-      show_criterion(parseInt(this.getAttribute('data-id'), 10), jQuery(this).attr('data-scheme'));
```

Comment:
@GitGeminiano I realized that `id` does not uniquely identify criteria, if we allow multiple types of criteria to appear. This means that `show_criterion` and `hide_criterion` need to have both an id and class passed in.

However, you do *not* need to have a new data attribute (`criterion-class`) to store this information. You should be able to use the existing (HTML) classes of the div themselves (either `mark_criterion_level_container` or the one for flexible criteria), and simply pass those into these functions, and do the corresponding search in `show_criterion` and `hide_criterion`.